src/jit/codegencommon.cpp

```diff
@@ -3942,7 +3942,8 @@ void            CodeGen::genFnPrologCalleeRegArgs(regNumber xtraReg,
                 
                 if ((!doingFloat &&
                     ((structDesc.eightByteClassifications[slotCounter] == SystemVClassificationTypeInteger) || 
-                     (structDesc.eightByteClassifications[slotCounter] == SystemVClassificationTypeIntegerReference))) ||
+                     (structDesc.eightByteClassifications[slotCounter] == SystemVClassificationTypeIntegerReference) ||
```

Comment:
Agree with the above comment.  It would be a good idea to add abstractions for each of the classification types.